x-pack/plugins/fleet/server/services/setup.ts

```diff
@@ -77,6 +82,15 @@ async function createSetupSideEffects(
   // By moving this outside of the Promise.all, the upgrade will occur first, and then we'll attempt to reinstall any
   // packages that are stuck in the installing state.
   await ensurePackagesCompletedInstall(soClient, callCluster);
+  if (appContextService.getConfig()?.agents.fleetServerEnabled) {
+    await ensureInstalledPackage({
+      savedObjectsClient: soClient,
+      pkgName: FLEET_SERVER_PACKAGE,
+      callCluster,
+    });
+    await ensureFleetServerIndicesCreated(esClient);
+    await runFleetServerMigration();
```

Comment:
I see these logs every time the Fleet UI loads. I wonder if additional conditions should be added here, or inside `runFleetServerMigration` to avoid this?

```
server    log   [10:46:10.727] [info][fleet][plugins] Starting fleet server migration
server    log   [10:46:10.730] [info][fleet][plugins] Fleet server migration finished
server    log   [10:47:45.611] [info][fleet][plugins] Starting fleet server migration
server    log   [10:47:45.613] [info][fleet][plugins] Fleet server migration finished
server    log   [10:48:26.913] [info][fleet][plugins] Starting fleet server migration
server    log   [10:48:26.915] [info][fleet][plugins] Fleet server migration finished
```

Reply:
We probably can remove the log, it do not really had any value

Reply:
Yes we are checking if there is saved object to migrate, this whole PR is temporary and just for allowing to test the whole flow in https://github.com/elastic/kibana/pull/89372 as we are going to move away from the package and put this in Kibana start.